src/codegen/test.jl

```diff
@@ -78,11 +82,11 @@ transpile(scope, inferredExpr)
 
 
 function cast_kernel(x::WgpuArray{T, N}, out::WgpuArray{S, N}) where {T, S, N}
-	xdim = workgroupDims.x
-	ydim = workgroupDims.y
-	gIdx = workgroupId.x*xdim + localId.x
-	gIdy = workgroupId.y*ydim + localId.y
-	gId = xDims.x*gIdy + gIdx
+	xdim::UInt32 = workgroupDims.x
```

Comment:
This change forces that the new variables are always declaration statements. Not ideal. But temporarily acceptable.